File: app/src/main/java/com/remotearthsolutions/expensetracker/services/FacebookService.java
package com.remotearthsolutions.expensetracker.services;

import com.facebook.CallbackManager;

public interface FacebookService {

    void facebookCallbackInitialize();
    void startFacebookLogin(CallBack callBack);
    CallbackManager getFacebookCallbackManager();


    interface CallBack extends SocialLoginCallback{
        void onFacebookLoginCancel();
    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/services/FacebookService.kt
package com.remotearthsolutions.expensetracker.services

import com.facebook.CallbackManager

interface FacebookService {
    fun facebookCallbackInitialize()
    fun startFacebookLogin(callBack: CallBack?)
    val facebookCallbackManager: CallbackManager?

    interface CallBack : SocialLoginCallback {
        fun onFacebookLoginCancel()
    }
}